### ISSUE TITLE
Add optimizer selection

### ISSUE THREAD

**Issue by aymuos15** [aymuos15/AutoMONAI#8]
Training is currently hardcoded to Adam. Add support for selecting the optimizer.

- **Adam** (default)
- **AdamW** — Adam with decoupled weight decay
- **SGD** — with momentum option

Selectable via `--optimizer adam|adamw|sgd` and exposed in both the web UI and TUI.
